tests/test_grid.py

```diff
@@ -74,34 +75,65 @@ def test_linear_grid(self):
     def test_linear_grid_spacing(self):
         """Test linear grid spacing is consistent."""
 
-        def test(endpoint=False, axis=True):
+        def test(sym, endpoint, axis, ntheta_is_odd):
             nrho = 1
-            ntheta = 5
+            # more edge cases are caught when ntheta is odd when endpoint=True
+            # and ntheta is even when endpoint is False
+            ntheta = 6 + ntheta_is_odd
             nzeta = 7
             NFP = 3
-            grid1 = LinearGrid(
+            theta = np.linspace(0, 2 * np.pi, ntheta, endpoint=endpoint)
+            if sym:
+                # When users supply nodes and set symmetry to true, they need
+                # to ensure their nodes are placed symmetrically.
+                # Here we change the array to match the effect of line 481
```

Comment:
We should avoid referencing specific line numbers, since those will change as the file is edited in future PRs.